### PR TITLE
refactor: collect java field and method lookups

### PR DESCRIPTION
**Pull request by JonathanStarup**
Begin collecting interop logic in the new `Jvm` object

### CODE REVIEW

File: main/src/ca/uwaterloo/flix/language/phase/Jvm.scala [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    *
+    * If the given class is an array, the method `clone` is not included (see Class.getMethods).
+    *
+    * Returns `Nil` if both static` and `instance` is `false`.
+    *
+    * @param clazz the class to search
+    * @param static whether to include static methods
+    * @param instance whether to include instance methods
+    */
+  def getMethodsHelper(clazz: Class[?], static: Boolean, instance: Boolean): List[Method] = {
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Jvm.scala:59]: private? name?

File: main/src/ca/uwaterloo/flix/language/phase/Jvm.scala [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  def getField(clazz: Class[?], fieldName: String, static: Boolean): Option[Field] = {
+    try {
+      val field = clazz.getField(fieldName)
+      if (isStatic(field) == static) Some(field) else None
+    } catch {
+      case _: NoSuchFieldException => None
+    }
+  }
+
+  /** Returns `true` if `member` is a static member ([[Field]] and [[Method]] extends [[Member]]). */
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Jvm.scala:40]: Can we not do normal doc comments?

File: main/src/ca/uwaterloo/flix/language/phase/Jvm.scala [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+ * limitations under the License.
+ */
+
+package ca.uwaterloo.flix.language.phase
+
+import java.lang.reflect.{Field, Member, Method}
+
+object Jvm {
+
+  /**
```
Comment (magnus-madsen) [inline main/src/ca/uwaterloo/flix/language/phase/Jvm.scala:23]: In general, how are functions ordered in this file?

File: main/src/ca/uwaterloo/flix/language/phase/Jvm.scala (hunk 2 above)
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/Jvm.scala:40]: what part of it do you mean? I didn't know `Member` existed so I though the clarification was nice

Comment (magnus-madsen) [reply main/src/ca/uwaterloo/flix/language/phase/Jvm.scala:40]: I mean not to use singleline docs. Your new found style which is not used anywhere else :)

**Comment by magnus-madsen** [comment]
Conflicted? RFR?

File: main/src/ca/uwaterloo/flix/language/phase/Jvm.scala (hunk 2 above)
Comment (JonathanStarup) [reply main/src/ca/uwaterloo/flix/language/phase/Jvm.scala:40]: > which is not used anywhere else :)

The bane of innovation and improvement! ✊

I can change it - but I do think its better

**Comment by JonathanStarup** [comment]
@magnus-madsen rfr

**Comment by magnus-madsen** [comment]
How about moving Jvm to one of the utils packages?

**Review by magnus-madsen — APPROVED** [review]

**Comment by magnus-madsen** [comment]
Request follow up PR to move Jvm to util package since it is not a phase.

**Pull request merged**